gk/fib.c

```diff
@@ -1881,6 +1881,186 @@ l_list_gk_fib6(lua_State *l)
 	return 1;
 }
 
+static void
+fillup_gk_neighbor_dump_entry(struct gk_neighbor_dump_entry *dentry,
+	struct ether_cache *eth_cache)
+{
+	dentry->stale = eth_cache->stale;
+	rte_memcpy(&dentry->neigh_ip, &eth_cache->ip_addr,
+		sizeof(dentry->neigh_ip));
+	rte_memcpy(&dentry->d_addr, &eth_cache->l2_hdr.eth_hdr.d_addr,
+		sizeof(dentry->d_addr));
+}
+
+#define CTYPE_STRUCT_NEIGHBOR_DUMP_ENTRY_PTR "struct gk_neighbor_dump_entry *"
+
+static void
+list_hash_table_neighbors(lua_State *l, enum gk_fib_action action,
+	struct neighbor_hash_table *neigh_ht)
+{
+	uint32_t next = 0;
+	const void *key;
+	void *data;
+	int32_t index;
+
+	static bool assigned_type_neighbor_dentry = false;
+	static uint32_t correct_ctypeid_neighbor_dentry;
+
+	void *cdata;
+
+	if (!assigned_type_neighbor_dentry) {
+		correct_ctypeid_neighbor_dentry = luaL_get_ctypeid(l,
+			CTYPE_STRUCT_NEIGHBOR_DUMP_ENTRY_PTR);
+		assigned_type_neighbor_dentry = true;
+	}
+
+	index = rte_hash_iterate(neigh_ht->hash_table,
+		(void *)&key, &data, &next);
+	while (index >= 0) {
+		struct gk_neighbor_dump_entry dentry;
+		struct ether_cache *eth_cache = &neigh_ht->cache_tbl[index];
+
+		dentry.action = action;
+		fillup_gk_neighbor_dump_entry(&dentry, eth_cache);
+
+		lua_pushvalue(l, 2);
+		lua_insert(l, 3);
+		cdata = luaL_pushcdata(l, correct_ctypeid_neighbor_dentry,
+			sizeof(struct gk_neighbor_dump_entry *));
+		*(struct gk_neighbor_dump_entry **)cdata = &dentry;
+		lua_insert(l, 4);
+
+		lua_call(l, 2, 1);
+
+		index = rte_hash_iterate(neigh_ht->hash_table,
+			(void *)&key, &data, &next);
+	}
+}
+
+static void
+list_ipv4_if_neighbors(lua_State *l,
+	struct gatekeeper_if *iface, struct gk_lpm *ltbl)
```

Comment:
1. Add parameter `enum gk_fib_action action` as described in other comments.
2. Do the same to `list_ipv6_if_neighbors()`.